besu/src/main/java/org/hyperledger/besu/cli/options/stable/DataStorageOptions.java

```diff
@@ -69,20 +69,26 @@ public class DataStorageOptions implements CLIOptions<DataStorageConfiguration>
   public static final String BONSAI_TRIE_LOG_PRUNING_WINDOW_SIZE =
       "--bonsai-trie-logs-pruning-window-size";
 
+  // TODO --Xbonsai-limit-trie-logs-enabled and --Xbonsai-trie-log-pruning-enabled are deprecated,
+  // remove in a future release
   @SuppressWarnings("ExperimentalCliOptionMustBeCorrectlyDisplayed")
   @CommandLine.Option(
       names = {
         BONSAI_LIMIT_TRIE_LOGS_ENABLED,
-        "--Xbonsai-limit-trie-logs-enabled",
-        "--Xbonsai-trie-log-pruning-enabled"
+        "--Xbonsai-limit-trie-logs-enabled", // deprecated
+        "--Xbonsai-trie-log-pruning-enabled" // deprecated
```

Comment:
maybe also hide the deprecated names so they don't appear in help?

Reply:
Can this be done easily?

I think I'd have to split into separate options to use the `hidden` attribute, then manually combine them all outside of the picocli framework upon loading. Not sure it's worth the effort.
I think they've been deprecated for ~3 months, so will probably remove them in a release or two anyway.

Reply:
yeah you would have to split it up. fair enough.